API/API/ItemComponent/Controllers/ItemController.cs

```diff
@@ -0,0 +1,126 @@
+﻿using API.Data;
+using API.ItemComponent.Models;
+using API.ItemComponent.Repository;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.ItemComponent.Controllers;
+// https://www.pragimtech.com/blog/blazor/delete-in-asp.net-core-rest-api/
+[ApiController]
+[Route("[controller]")]
+
+public class ItemController : ControllerBase
+{
+    private readonly ILogger<ItemController> _logger;
+    private readonly IItemRepository _itemRepository;
+    
+    public ItemController(ILogger<ItemController> logger, IItemRepository itemRepository)
+    {
+        _logger = logger;
+        _itemRepository = itemRepository;
+    }
+    [HttpPost("Item")]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    
+    public async Task<ActionResult<Item>> CreateItem(Item? item)
+    {
+        try
+        {
+            if (item == null)
+            {
+                return BadRequest();
+            }
+
+            var createdItem = await _itemRepository.AddItem(item);
+
+            return CreatedAtAction(nameof(GetItems), new { id = createdItem.id }, createdItem);
+        }
+        catch (Exception e)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError, "Error retrieving data from the database");
+        }
+    }
+    
+    [HttpGet("Item/Items")]
+    public async Task<ActionResult<List<Item>>> GetItems()
+    {
+        try
+        {
+            return Ok(await _itemRepository.GetItem());
+        }
+        catch (Exception)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError,
+                "Error retrieving data from the database");
+        }
+    }
+    
+    [HttpGet("Item/{id}")]
+    public async Task<ActionResult<Item>> GetItem(Guid id)
+    {
+        try
+        {
+            var result = await _itemRepository.GetItem(id);
+
+            if (result == null)
+            {
+                return NotFound();
```

Comment:
Reiktu kazkokio pranesimo klientui. Galima pakeisti į NotFound($"Item with Id = {id} not found");